package/scripts/precook_test.py

```diff
@@ -0,0 +1,169 @@
+            " ".join(
+                [
+                    str(random.randint(1, time_lim)) if rand else str(fixed_value)
+                    for i in range(4)
+                ]
+            ),
+            file=f,
+            end=" ",
+        )
+
+        # 17 and 18 are never used -> set to 0
+        print("0 0", file=f, end=" ")
+
+        # 19 num_threads - Number of threads of the process
+        num_threads = 1
+        print(num_threads, file=f, end=" ")
+
+        # 20 never used -> set to 0
+        print("0", file=f, end=" ")
+
+        # 21 starttime (uptime in utils.h) (last field that is read by any monitor)
+        print(random.randint(1, time_lim) if rand else fixed_value, file=f)
+
+
+def make_io(proc_pid, fixed_value, rand=False):
+    io_fname = os.path.join(proc_pid, "io")
+    fields = [
+        "rchar: ",
+        "wchar: ",
+        "syscr: ",
+        "syscw: ",
+        "read_bytes: ",
+        "write_bytes: ",
+        "cancelled_write_bytes: ",
+    ]
+    char_lim = 1000
+    with open(io_fname, "w") as f:
+        for field in fields:
+            print(
+                f"{field}{random.randint(0, char_lim) if rand else fixed_value}", file=f
+            )
+
+
+def make_smaps(proc_pid, fixed_value, rand=False):
+    smaps_fname = os.path.join(proc_pid, "smaps")
+    fields = ["Size: ", "Pss: ", "Rss: ", "Swap: "]
+    mem_lim = 1000
+    with open(smaps_fname, "w") as f:
+        for field in fields:
+            print(
+                f"{field}{random.randint(0, mem_lim) if rand else fixed_value} kB",
+                file=f,
+            )
+
+
+def make_net(proc_net, fixed_value, rand=False):
+    net_params = ["rx_bytes", "tx_bytes", "rx_packets", "tx_packets"]
+    net_lim = 1000000
+    for param in net_params:
+        net_fname = os.path.join(proc_net, param)
+        with open(net_fname, "w") as f:
+            print(random.randint(0, net_lim) if rand else fixed_value, file=f)
+
+
+def make_nvidia(proc_nvidia, fixed_value):
+    # idx
+    smi_fname = os.path.join(proc_nvidia, "smi")
+    with open(smi_fname, "w") as f:
+        params = [
+            0,  # idx
+            pid,  # pid
+            "G",  # type
+            fixed_value,  # sm
+            fixed_value,  # mem
+            # enc, dec are not monitored metrics
+            0,  # enc
+            0,  # dec
+            fixed_value,  # fb
+            "python3",  # command
+        ]
+        for param in params:
+            print(param, file=f, end=" ")
+        print("", file=f)
+
+
+def createPath(new_dir):
+    dir_name = os.path.join(os.getcwd(), "precooked_tests")
+    if not os.path.isdir(dir_name):
+        os.mkdir(dir_name)
+    dir_name = os.path.join(dir_name, new_dir)
+    if os.path.isdir(dir_name):
+        shutil.rmtree(dir_name)
+    os.mkdir(dir_name)
+    return dir_name
+
+
+def createTestMonotonic():
+    # 3 iterations having first increasing and then decreasing stats
+    dir_name = createPath("drop")
```

Comment:
It would be nice if this could be configured on the command line - use the `argparse` module to do that (tons of examples around too, e.g., 

https://github.com/HSF/prmon/blob/d1f512fd3e17182bbd9b4c45e1e57e0ea4d6dea5/package/scripts/prmon_plot.py#L134-L140

Reply:
Do you mean the number of iterations or the name of the directory created?

Reply:
Why not both! Once you add `argparse` customisation becomes very easy and both could be useful for any future set of pre-cooked values.